server/src/main/java/app/user/UserService.java

```diff
@@ -19,9 +24,26 @@ public User getUserById(Integer userId) {
                 .orElseThrow(() -> new UsernameNotFoundException("User not found with id: " + userId));
     }
 
-    public List<UserSearchResponse> searchUsers(String query) {
-        Pageable pageable = PageRequest.of(0, 5); // Pierwsza strona, 5 wyników
-        return userRepository.findByNameContainingIgnoreCase(query, pageable).stream()
+    public ResponseEntity<UserProfileResponse> getUserProfile(Integer userId, UserDetails userDetails) {
+        if (!userDetails.getAuthorities().stream()
+                .anyMatch(auth -> auth.getAuthority().equals("ADMIN") || auth.getAuthority().equals("USER"))) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+            }

```

Comment:
możesz to teraz usunąć. zpulluj sobie zmiany mastera

git checkout main
git pull origin 
git checkout kacper
git rebase main

zobacz security config - tam są nasze endpointy i opisane zasady Cors, nie będziesz mógł się połączyć z endpointami UserController bez JwtTokena,